packages/lib/services/rest/routes/notes.test.ts

```diff
@@ -76,6 +76,44 @@ describe('routes/notes', () => {
 
 		const response = await downloadMediaFile(url);
 
+		const files = await readdir(Setting.value('tempDir'));
+		expect(files.length).toBe(0);
 		expect(response).toBe('');
 	});
+
+	test('should not copy content from invalid protocols', async () => {
+		const url = 'file:///home/user/file.db';
+
+		const allowedProtocols: string[] = [];
+		const mediaFilePath = await downloadMediaFile(url, null, allowedProtocols);
+
+		const files = await readdir(Setting.value('tempDir'));
+		expect(files.length).toBe(0);
+		expect(mediaFilePath).toBe('');
+	});
+
+	test.each([
+		'https://joplinapp.org/valid/image_url',
+		'https://joplinapp.org/valid/image_url.invalid_url',
+	])('should find and move file with invalid or without filename', async (url) => {
+		const spy1 = jest.spyOn(shim, 'fetchBlob').mockImplementation(() => {
+			return {
+				headers: {
+					'content-type': 'image/jpg',
+				},
+			};
+		});
+		const shimFsDriverMoveSpy = jest.fn();
+		const spy2 = jest.spyOn(shim, 'fsDriver').mockImplementation(() => {
+			return {
+				move: shimFsDriverMoveSpy,
+			} as any;
+		});
```

Comment:
The spy shouldn't be needed since you're mocking fetchBlob and can make it create real files?